### ISSUE TITLE
Fix EPSG:3006 support

### ISSUE THREAD

**Issue by muttcg** [gbif/pipelines#456]
Coordinate parser checks range first and after tries to apply reprojection, which is not suitable for coordinate system like https://epsg.io/3006:

```
private static boolean inRange(double lat, double lon) {
    return Double.compare(lat, 90) <= 0
        && Double.compare(lat, -90) >= 0
        && Double.compare(lon, 180) <= 0
        && Double.compare(lon, -180) >= 0;
  }
```

[Dataset example](https://www.gbif.org/dataset/252f45fe-dfb6-46d6-9457-59bb3947c517). Record fragment:
```
  "decimalLatitude": "6911438.25017224159",
  "decimalLongitude": "627186.764745077118",
  "geodeticDatum": "EPSG:3006",
```

**Comment by jhnwllr** [comment]
Wondering what @tucotuco opinion is about this issue... 

Since 
https://dwc.tdwg.org/terms/#dwc:decimalLatitude

> The geographic latitude (in decimal degrees, using the spatial reference system given in geodeticDatum) of the geographic center of a Location. Positive values are north of the Equator, negative values are south of it. Legal values lie between -90 and 90, inclusive.

Hard to communicate the standard while also not being too technical.

**Comment by tucotuco** [comment]
The way these original data were mapped is incorrect, but it seems a shame to throw good data away. I mean, how many coordinates do we get that are actually meaningless because they don't include a coordinate reference system? The correct mapping would have been to verbatimLatitude, verbatimLongitude, verbatimSRS.

Are there a lot of other examples where a geodeticDatum is provided that corresponds to a CRS that does not use geographic coordinates? The potential number is large, but it should be possible to detect that and skip the decimal coordinate validation. The problem comes if you want to validate the range for every possibility. But then, what happens if the mapping is actually correct (e,g,  to verbatimLatitude, verbatimLongitude, verbatimSRS) for this case?

**Comment by anneliejonsson** [comment]
Thank you all for your input. I'm responsible for the original faulty mapping. I basically used the same set of dwc-terms that I had used previously on a new dataset, but forgot to think about the "format" of the coordinates in the new dataset. To start with, using decimalLatitude/Longitude kind of hints that your coordinates need to be in decimal format... But I have now read the definitions for all the terms properly so I know which to use when in the future. We have yet to try it though. Anyhow, thanks again. /Annelie

**Comment by tucotuco** [comment]
Hi @anneliejonsson,

The term name verbatimCoordinateSystem is unfortunately misleading. It should better be called verbatimCoordinateFormat. In the case of the Coordinate Reference System epsg:3006 the verbatimCoordinateSystem should be epsg:4500, which translates in human-readable terms to "Cartesian 2D CS. Axes: northing, easting (N,E). Orientations: north, east. UoM: m."

Cheers,

John

**Comment by tucotuco** [comment]
I'm curious what it is that doesn't work?  Your epsg:3006 coordinates in the verbatim coordinate fields do not get translated into geographic coordinates in epsg:4326 by the GBIF pipeline?

**Comment by anneliejonsson** [comment]
Hi there,
Thanks for your replies! I'll ask my colleague to add the fourth field with epsg:4326 and see if that makes a difference. But to answer your question: That's right. gbif doesn't "pick up" the coordinates we've supplied in the event table. The dataset we're talking about has the DOI 10.15468/37u448" (Coastal birds in the Gulf of Bothnia). As you can see there we get 0% occurrences with coordinates...

**Comment by timrobertson100** [comment]
I think this happens because even though the verbatim values are [read](https://github.com/gbif/pipelines/blob/dev/sdks/core/src/main/java/org/gbif/pipelines/core/parsers/location/parser/CoordinatesParser.java#L59) we still filter for within range 90/180 [here](https://github.com/gbif/pipelines/blob/dev/sdks/core/src/main/java/org/gbif/pipelines/core/parsers/location/parser/CoordinateParseUtils.java#L94).

I think we need to relax that restriction when the verbatim values are read, and this now looks like a bug (or perhaps more accurately, previously unspecified behaviour)

**Comment by anneliejonsson** [comment]
Thanks Tim! Do you know if that's likely to be "fixed" in the near future, or should we use a different coordinate reference system for now? We have successfully uploaded other datasets where we used wsg84 instead.

**Comment by timrobertson100** [comment]
Shouldn't be too long unless some other quirk appears, but it may not be within the next week or two.

**Comment by anneliejonsson** [comment]
All right, thanks. Might change it to wgs84 for now. The reason for using epsg:3006 in the first place is that it's "demanded" in our contract. Nevermind. Better to have coordinates that show. Many thanks for your input!

**Comment by MattBlissett** [comment]
One way to satisfy both GBIF and your contract could be to put the EPSG:3006 coordinates in the `verbatim*` terms, and WGS84 coordinates in `decimalLatitude` etc.

**Comment by anneliejonsson** [comment]
Yes of course! I get lost among all these terms... I was thinking I couldn't have two different types of coordinates but of course I can since they are mapped to different dwc-terms. Thanks for pointing that out!
